Modified DataCharger minigame

Ekiz24 opened the pull request.

Ekiz24 committed:
Created and modified 2 personal items

1. Name Tag on the desk
2. Name on the electricity bill

Ekiz24 committed:
Replace the personal book (1/4 images)

The front view of the personal book: 'Chronicle of Pharaohs' to 'Game of Shades of Gray'

Ekiz24 committed:
Modified instructions panel

1. Created a new instruction
2. Changed the size and position of the panel
3. Changed its order to make instructions panel be on top of the table(playing screen)

Ekiz24 committed:
BGM Controller

1. Added icons for music on/off
2. Wrote scripts to control music on/off and icon switching

Ekiz24 committed:
Changed the style of the nametag

1. Created a new nameplate, deleted the old one.
2. Changed the position of a book to make the name tag fit.

Ekiz24 committed:
Replaced the personal book

1. Deleted old images
2. Created images of left and right side view
3. The corresponding substitutions were made in the Inspector

Ekiz24 committed:
Created and show particles

1. Changed render mode of all canvas to Screen Space - Camera.(to show particles properly)
2. Added a sprite of particle effect
3. Created a material for particles
4. Adjusted parameters of components in Particle System

Ekiz24 committed:
Improved particles

1. Recorded animation of particles, and set trigger conditions.
2. Updated the scripts SortBoxBehaviour to show different particles animation under different situation.

Ekiz24 committed:
Improved Particles

1. Created discharge animation
2. Modified particle parameters/style/effects
3. Updated the script SortBoxBehaviour to show animations of discharge

Ekiz24 committed:
Fixed 2 small bugs

1. Changed the original position of particles.
2. Changed the personal book image in the Win-Screen

AShenawy committed:
Merge branch 'Development' into MiniGame-Sorting

AShenawy committed:
Update Data Charger.unity

Fix indentation from previous merge commit

AShenawy commented:
Fixed scene coflicts

AShenawy committed:
Update EditorBuildSettings.asset

Revert disabled scenes

AShenawy commented on Assets/Minigames/Sorting/Scripts/Sound/MusicController.cs:
```diff
@@ -0,0 +1,33 @@
+
+    public void ToggleMusic()
+    {
+        buttonImage = GetComponent<Image>();
```
Don't use GetComponent if you can assign it in the inspector. And definitely don't GetComponent every time the function is called. If you need to use it then do that once in Awake or Start and cache the reference in a variable (buttonImage which you're already using)

In this case, assign the component to the variable in the Inspector. If you want to keep the variable private then add `[SerializeField]` to it
```cs
[SerializeField]
private Image buttonImage;
```

AShenawy commented on Assets/Minigames/Sorting/Scripts/Sound/MusicController.cs:
```diff
@@ -0,0 +1,33 @@
+    public void ToggleMusic()
+    {
+        buttonImage = GetComponent<Image>();
+        AudioSource backgroundMusic = audioManager.GetComponentInChildren<AudioSource>();
```
If I remember correctly, the audiomanager object has multiple child audio sources. How can you tell that the one you got is the one responsible for the background music?
Also, there's a SoundManager script/object in the scene that you should use to add the muting functionality to, not here. This way the functionality will be used for the entire game and not just one minigame - keeps things consistnet

Ekiz24 replied:
Hi Shen, I have some confusion here.

1. Is the old 'Data Charger.unity' I was working on before the current 'Data Charger Standalone.unity'? Because before there was only one scene and in that scene there was 'Camera' and 'Audio Manager (with a script SoundManager)', now the current 'Data Charger.unity' doesn't have them.

2. There is only 1 audio source under 'AudioManager' Object for background music, and the other sound effects are in Charge Station QN/QL. (Only in the old 'Data Charger.unity' where I worked before and now in the current 'Data Charger Standalone.unity')

3. I was testing the current 'Data Charger Standalone.unity' and found that some of the original basic functions are not working, but I don't know why.

4. I'll rework the code for controlling the BGM and add it to the SoundManager script, but I'm not sure how to control the BGM in the current ‘Data Charger.unity’ yet, as it doesn't have the 'Audio Manager Object' in it as well as the 'Sound Manager Script'.

Ekiz24 replied:
Just a side note, if this isn't an easy fix, I can redo all my history work on the correct version.

AShenawy replied:
Hey Zike,

1. The standalone has them because that's the minigame on its own without the full game, so that you can play it in the minigames section of the [webpage ](https://dlg.tlu.ee/methodyca/mini-games/data-charger).
In the full game, the Sound Manager isn't in that scene because the Data Charger scene is loaded on top of the game world scene. And the Sound Manager is in that world scene already.

2. If you look at SoundManager script and in the PlaySFX function, every time a new SFX is played a new AudioSource component is added to Sound Manager game object. `GetComponentInChildren()` will in practice look at the game object itself first before its child objects and thus will return to you the SFX components - especially so if there's a looping sfx. Check out the `Sound` script as well in the same folder of the SoundManager script.

3. I don't know what you mean by "original basic functions" you have to clarify and better on discord chat.

4. Refer to point 1

You don't have to redo anything. Just modify and update the current situation. The sorting minigame branch has also been updated now to match with Development branch

AShenawy reviewed (changes requested):
Please check the comments and make the necessary changes

Ekiz24 committed:
Modified BGM control

1. Deleted the script MusicController
2. Added ToggleMusic function to the script SoundManager
3. Added the script SoundManager to the object BGM on/off Button

Ekiz24 committed:
Changed the personal book

1. Replaced previous images with new images
2. Changed sprites in the prefabs of the personal book

AShenawy reviewed (approved):
Looks fine to me. Just one note: the muting functionality currently doesn't technically mute, but instead pause the music playback.
